pkg/watermark/fetch/processor_to_fetch.go

```diff
@@ -58,9 +58,8 @@ func NewProcessorToFetch(ctx context.Context, processor processor.ProcessorEntit
 		otWatcher:      watcher,
 		log:            logging.FromContext(ctx),
 	}
-	if watcher != nil {
```

Comment:
any particular reason why you did this? (ie, moved to `startTimeLineWatcher`)?

Reply:
My thought behind this change is that, since `watcher` is a member of ProcessorToFetch p, so it should be p's responsibility to do NULL check on watcher. By moving this check to ProcessorToFetch, we ensure ProcessorToFetch never start the watcher if watcher is NULL and on the other hand, all future callers on startTimeLineWatcher don't need to worry about watcher being NULL.